host_entry.go

```diff
@@ -42,25 +43,49 @@ func (h *hostEntry) willRetryHost(maxRetryInterval time.Duration) {
 }
 
 func (h *hostEntry) getWeightedAverageResponseTime() float64 {
-	var value float64
-	var lastValue float64
+	currentBucketCount := h.epsilonCounts[h.epsilonIndex]
```

Comment:
I added some test cases in `host_entry_test.go` over in [this branch](https://github.com/monzo/go-hostpool/compare/master...monzo:kjgorman/configurables?expand=1) where I was extracting some properties (like bucket counts) — they highlight that in a zero case of no hosts this line can panic

Reply:
This panic will occur if the hostEntry hasn't been set up properly. if we look at where it is set up, [it is initialised with the zero values for all the buckets it expects](https://github.com/monzo/go-hostpool/blob/master/epsilon_greedy.go#L65-L69).

At this point, I feel like an internal invariant has been broken and it _should_ panic rather than paper over the crack by adding a condition within the calculation function.